### PR TITLE
build: automated versioning and publishing

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  release:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Use Node.js 18.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: "16.x"
```
Comment (rsdmike) [inline .github/workflows/release.yml:24]: this should be 18

File: package.json [hunk 2]
```diff
@@ -73,6 +73,7 @@
     "jest": "27.5.1",
     "jest-junit": "^16.0.0",
     "jest-transform-stub": "^2.0.0",
+    "semantic-release": "^21.0.2",
```
Comment (rsdmike) [inline package.json:76]: not needed since we run in CI

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (madhavilosetty-intel) [reply .github/workflows/release.yml]: done

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      uses: actions/setup-node@v3
+      with:
+        node-version: "18.x"
+    - run: npm ci
+    - run: npm run build-ext --if-present
+    - run: rsync -a package.json README.md ./dist/
+    - name: Semantic Release
+      uses: cycjimmy/semantic-release-action@v3
+      with:
+        semantic_version: 19.0.5  # It is recommended to specify a version range
```
Comment (rsdmike) [inline .github/workflows/release.yml:31]: is this the latest version?

Comment (madhavilosetty-intel) [reply .github/workflows/release.yml:31]: Yes, as per the link https://github.com/cycjimmy/semantic-release-action I updated it.

**Review by matt-primrose — APPROVED** [review]

**Review by rsdmike — APPROVED** [review]

**Pull request merged**